app/models/manageiq/providers/openstack/inventory/collector/cloud_manager.rb

```diff
@@ -144,4 +144,22 @@ def orchestration_resources(stack)
   def orchestration_template(stack)
     safe_call { stack.template }
   end
+
+  def volume_templates
+    return [] unless volume_service
+    return @volume_templates if @volume_templates.any?
+    @volume_templates = volume_service.handled_list(:volumes, {:status => "available"}, ::Settings.ems.ems_openstack.refresh.is_admin)
+  end
+
+  def volumes_by_id
+    @volumes_by_id ||= Hash[volume_templates.collect { |f| [f.id, f] }]
```

Comment:
Alternately, you can use the index_by method.

```ruby
@volumes_by_id ||= volume_templates.index_by(&:id)
```

Reply:
Thanks for that shortcut!